### ISSUE TITLE
Exception when validating a PDF

### ISSUE THREAD

**Issue by tlilja** [MatthiasValvekens/pyHanko#75]
**Describe the bug**

Trying to validate a PDF with `pyhanko sign validate` and I got a Traceback with KeyError.

**To Reproduce**

Execute
```
pyhanko sign validate --pretty-print  signed-lorem-ipsum.pdf
```
I got:
```
022-02-03 14:54:47,419 - pyhanko.cli - ERROR - Generic processing error.
Traceback (most recent call last):
  File "$VENV/lib/python3.8/site-packages/pyhanko_certvalidator/fetchers/common_utils.py", line 164, in queue_fetch_task
    wait_event: asyncio.Event = running_jobs[tag]
KeyError: 'https://www.sk.ee/upload/files/TEST_of_EE_Certification_Centre_Root_CA.der.crt'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "$VENV/lib/python3.8/site-packages/pyhanko/cli.py", line 78, in pyhanko_exception_manager
    yield
  File "$VENV/lib/python3.8/site-packages/pyhanko/cli.py", line 529, in validate_signatures
    (status_str, signature_ok) = _signature_status_str(
  File "$VENV/lib/python3.8/site-packages/pyhanko/cli.py", line 387, in _signature_status_str
    status = status_callback()
  File "$VENV/lib/python3.8/site-packages/pyhanko/cli.py", line 530, in <lambda>
    status_callback=lambda: _signature_status(
  File "$VENV/lib/python3.8/site-packages/pyhanko/cli.py", line 349, in _signature_status
    status = pyhanko.sign.validation.validate_pdf_signature(
  File "$VENV/lib/python3.8/site-packages/pyhanko/sign/validation/__init__.py", line 221, in validate_pdf_signature
    return asyncio.run(coro)
  File "/Users/tlilja/.pyenv/versions/3.8.10/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Users/tlilja/.pyenv/versions/3.8.10/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "$VENV/lib/python3.8/site-packages/pyhanko/sign/validation/pdf_embedded.py", line 845, in async_validate_pdf_signature
    ts_status_kwargs = await collect_timing_info(
  File "$VENV/lib/python3.8/site-packages/pyhanko/sign/validation/generic_cms.py", line 432, in collect_timing_info
    tst_validity_kwargs = await validate_tst_signed_data(
  File "$VENV/lib/python3.8/site-packages/pyhanko/sign/validation/generic_cms.py", line 492, in validate_tst_signed_data
    return await cms_basic_validation(
  File "$VENV/lib/python3.8/site-packages/pyhanko/sign/validation/generic_cms.py", line 270, in cms_basic_validation
    trusted, revoked, path = await status_cls.validate_cert_usage(
  File "$VENV/lib/python3.8/site-packages/pyhanko/sign/validation/status.py", line 154, in validate_cert_usage
    path = await validator.async_validate_usage(key_usage=set())
  File "$VENV/lib/python3.8/site-packages/pyhanko_certvalidator/__init__.py", line 263, in async_validate_usage
    await self._validate_path()
  File "$VENV/lib/python3.8/site-packages/pyhanko_certvalidator/__init__.py", line 107, in _validate_path
    paths = await self._context.certificate_registry.async_build_paths(
  File "$VENV/lib/python3.8/site-packages/pyhanko_certvalidator/registry.py", line 323, in async_build_paths
    await self._walk_issuers(path, paths, failed_paths)
  File "$VENV/lib/python3.8/site-packages/pyhanko_certvalidator/registry.py", line 375, in _walk_issuers
    async for issuer in self.fetcher.fetch_cert_issuers(path.first):
  File "$VENV/lib/python3.8/site-packages/pyhanko_certvalidator/fetchers/common_utils.py", line 268, in complete_certificate_fetch_jobs
    certs_fetched = await fetch_job
  File "$PYENV/versions/3.8.10/lib/python3.8/asyncio/tasks.py", line 619, in _wait_for_one
    return f.result()  # May raise f.exception().
  File "$VENV/lib/python3.8/site-packages/pyhanko_certvalidator/fetchers/requests_fetchers/cert_fetch_client.py", line 57, in fetch_certs
    return await self._perform_fetch(url, task)
  File "$VENV/lib/python3.8/site-packages/pyhanko_certvalidator/fetchers/requests_fetchers/util.py", line 37, in _perform_fetch
    return await queue_fetch_task(
  File "$VENV/lib/python3.8/site-packages/pyhanko_certvalidator/fetchers/common_utils.py", line 195, in queue_fetch_task
    return _return_or_raise(result)
  File "$VENV/lib/python3.8/site-packages/pyhanko_certvalidator/fetchers/common_utils.py", line 200, in _return_or_raise
    raise result
  File "$VENV/lib/python3.8/site-packages/pyhanko_certvalidator/fetchers/common_utils.py", line 182, in queue_fetch_task
    result = await async_fun()
  File "$VENV/lib/python3.8/site-packages/pyhanko_certvalidator/fetchers/requests_fetchers/cert_fetch_client.py", line 49, in task
    results = await self._grab_certs(
  File "$VENV/lib/python3.8/site-packages/pyhanko_certvalidator/fetchers/requests_fetchers/cert_fetch_client.py", line 103, in _grab_certs
    content_type = response.headers['Content-Type'].strip()
  File "$VENV/lib/python3.8/site-packages/requests/structures.py", line 54, in __getitem__
    return self._store[key.lower()][1]
KeyError: 'content-type'
Error: Generic processing error.
```

**Expected behavior**

No traceback. Perhaps error message or successful validation?

**Environment (please complete the following information):**

pyHanko==0.12.0
pyhanko-certvalidator==0.19.2

**Additional context**

Looks like https://www.sk.ee/upload/files/TEST_of_EE_Certification_Centre_Root_CA.der.crt is serving the file without `Content-Type` header and hence the error.

**Comment by MatthiasValvekens** [comment]
Yep, that looks like a bug in [pyhanko-certvalidator](https://github.com/MatthiasValvekens/certvalidator) alright. I'll fix it over there and do a bugfix release. Thanks for the report!

In the meantime, you should be able to work around this problem by downloading the relevant issuer certificate manually, and telling pyHanko about it using `--other-certs` (cf. my answer in #55, which has to do with a certificate fetching bug as well).

**Comment by MatthiasValvekens** [comment]
Fixed in `pyhanko-certvalidator==0.19.3`; thanks again for the report!

However, on closer inspection, the certificate you're trying to download looks like it's a root cert (judging from the URL). In that case, this fix should ensure that you don't get any nasty errors anymore, but it's not enough to actually get the signature to _validate_: for obvious reasons, pyHanko never assigns trust root status to certificates downloaded from the internet, so it will complain that the signer's certificate is untrusted.

In other words, you'll have to download the root cert you need---it looks like you're using test certificates, so no need to be particularly careful---and pass it to `pyhanko sign validate` using the `--trust` option.


Hope that helps :).
